src/App.tsx

```diff
@@ -1,223 +1,199 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useState,
+  useMemo,
+  useCallback,
+} from 'react';
+import classNames from 'classnames';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Todo } from './types/Todo';
+import {
+  getTodos,
+  addTodo,
+  removeTodo,
+} from './api/todos';
+import { FilterType } from './types/FilterType';
+import { ErrorNotification } from './types/ErrorNotification';
+import { NewTodo } from './components/NewTodo/NewTodo';
+import { TodoList } from './components/TodoList/TodoList';
+import { FilterTodos } from './components/FilterTodos/FilterTodos';
+import { Error } from './components/Error/Error';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState(FilterType.All);
+  const [title, setTitle] = useState('');
+  const [error, setError] = useState(ErrorNotification.None);
+  const [isAdding, setIsAdding] = useState(false);
+  const [loadTodosIds, setLoadTodosIds] = useState<number[]>([]);
 
-  useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+  const activeTodos = todos.filter(todo => !todo.completed);
+
+  const completedTodos = todos.filter(todo => todo.completed);
```

Comment:
you have switch case below why do you need this?

Reply:
For class changes and to provide the number of todos still active.